network/src/main/java/info/metadude/android/eventfahrplan/network/utils/DateHelper.kt

```diff
@@ -1,65 +0,0 @@
-@file:JvmName("DateHelper")
-
-package info.metadude.android.eventfahrplan.network.utils
-
-import android.text.format.Time
-import java.text.ParseException
-import java.text.SimpleDateFormat
-import java.util.*
-
-fun dateIsWithinRange(date: Date, dateRange: Array<Date>): Boolean {
-    val oldest = dateRange[0]
-    val newest = dateRange[1]
-    return (date == oldest || date.after(oldest)) && (date == newest || date
-            .before(newest))
-}
-
-/**
- * Returns a formatted date string.
- * This pattern is used: yyyy-MM-dd'T'HH:mm:ssZ
- */
-fun getFormattedDate(date: Date): String {
-    return getFormattedDate(date, "yyyy-MM-dd'T'HH:mm:ssZ")
-}
-
-/**
- * Returns a formatted date string.
- */
-fun getFormattedDate(date: Date, pattern: String): String {
-    val dateFormat = SimpleDateFormat(pattern, Locale.US)
-    return dateFormat.format(date)
-}
-
-fun getDateTime(text: String): Long {
-    val pattern = if (text.length > 10) "yyyy-MM-dd'T'HH:mm:ssZ" else "yyyy-MM-dd"
-    val date = getDate(text, pattern)
-    return date?.time ?: 0
-}
-
-fun getDayChange(attributeValue: String): Int {
-    val pattern = if (attributeValue.length > 10) "yyyy-MM-dd'T'HH:mm:ssZ" else "yyyy-MM-dd"
-    val date = getDate(attributeValue, pattern)
-            ?: return 600         // default
-    val timeUTC = date.time
-    val time = Time()
-    time.set(timeUTC)
-    return time.hour * 60 + time.minute
-}
```

Comment:
# Background information

I found a second usage of this value [`FahrplanParser.java:L122`](https://github.com/EventFahrplan/EventFahrplan/blob/master/network/src/main/java/info/metadude/android/eventfahrplan/network/serialization/FahrplanParser.java#L122). It originates from commit 1be19716c2982496526ccdba23b2d95089fbb49c which does not reveal much about why `600` was chosen.

Looking at `Lecture.Companion#parseStartTime` which is invoked in `ParserTask#parseFahrplan` to parse the value of the `<day_change>` node from the _schedule.xml_ I reconstructed the meaning of `600` representing the minutes of the day resp. 10 o'clock. This is when the first lecture took place.

In the _schedule.xml_ of 2012 (29C3) the `<day_change>` node was still present:

![Screenshot of the 29C3 schedule.xml header](https://user-images.githubusercontent.com/144518/75494076-ac23b880-59bb-11ea-9d58-b5d1f4743b0a.png)

https://fahrplan.events.ccc.de/congress/2012/Fahrplan/schedule.en.xml

In the _schedule.xml_ of 2013 (30C3) the node was missing. This might have been when the CCC changes their backend system from Pentabarf to Frab (assumption).

![Screenshot of the 30C3 schedule.xml header](https://user-images.githubusercontent.com/144518/75494153-e0977480-59bb-11ea-92fb-e8040f6ed064.png)

https://fahrplan.events.ccc.de/congress/2013/Fahrplan/schedule.xml

I inspected the start times of sessions in this year and found that the earliest was 11:00. By this I assume the programmer of that year [has chosen 10:00 (600)](https://github.com/EventFahrplan/EventFahrplan/commit/1be19716c2982496526ccdba23b2d95089fbb49c#r30436087) as a "good enough" time for the day change. - And there I rediscovered the very same question from myself asked in 2018 :)

I verified this theory with the [schedule.xml](https://fosdem.org/2020/schedule/xml) of FOSDEM 2020 which still uses Pentabarf as their backend which therefore provides `<day_change>09:00:00</day_change>`. The earliest session has taken place sharp at 9 o'clock.

:arrow_right: I hope this background information helps you when refactoring this.

Reply:
I wove the information / suggestions from our discussion into df92012e9e412ab19fcd251a60cc905ad36a5208. I am not sure if I got it all right though.

Reply:
Thank you. I close this here. I posted a corrected commit message in https://github.com/EventFahrplan/EventFahrplan/pull/245#discussion_r387837448.